src/App.tsx

```diff
@@ -1,5 +1,41 @@
-import React from 'react';
+import React, { useContext } from 'react';
+import { BrowserRouter as Router, Switch, Route, Redirect } from 'react-router-dom';
+import { AuthContext } from './context/AuthProvider';
+import ProtectedRoute from './ProtectedRoute';
 
-const App: React.FC = () => <h1>hello world</h1>;
+const App: React.FC = () => {
+  const user = useContext(AuthContext);
+
+  return (
+    <div>
+      <Router>
+        <Switch>
+          <ProtectedRoute exact path="/feed" />
+
+          <Route exact path="/" />
+
+          <Route exact path="/signUp" />
+
+          <Route exact path="/signIn" />
+
+          <Route
+            path="/"
+            render={() => {
+              if (user === undefined) {
```

Comment:
I think Protected Routes should handle this case. I see code duplication. You have to import authContext in 2 places and perform almost the same logic

Reply:
I know it's a duplication, but it is something like a default redirect, when the user enters anything after /, if he is not logged in, then he will be redirected to the welcom page, otherwise to his home with feed list
![scr](https://user-images.githubusercontent.com/87152087/126279723-23814530-9a5f-4780-b9d3-3b8634996b03.png)

Reply:
I know it's a duplication, but it is something like a default redirect, when the user enters anything after /, if he is not logged in, then he will be redirected to the welcom page, otherwise to his home with feed list

Reply:
and this is the only case without exact prop and it is at the very bottom so if none of the routes above match, this will be act like the default one